### ISSUE TITLE
[Bug]: Latest release Failed to compile application on  flutter version 3.19.5

### ISSUE THREAD

**Issue by santoshgistto** [hm21/pro_image_editor#117]
### Package Version

4.0.4

### Flutter Version

3.19.5

### Platforms

Web

### Logs

> 
../../../../.pub-cache/hosted/pub.dev/pro_image_editor-4.0.4/lib/utils/content_recorder.dart/content_recorder_controller.dart:402:9: Error: No named parameter with the name 'logicalConstraints'.
        logicalConstraints: BoxConstraints(
        ^^^^^^^^^^^^^^^^^^
../../../../flutter/packages/flutter/lib/src/rendering/view.dart:23:9: Context: Found this candidate, but the arguments don't match.
  const ViewConfiguration({
        ^^^^^^^^^^^^^^^^^
Failed to compile application.

_No response_

**Comment by hm21** [comment]
This is not an issue with the image editor, it's just one of the breaking changes in Flutter. 
The newest image editor version is compatible with the newest version of flutter `3.22.2`. I recommend upgrading to the latest version of flutter by running the `flutter upgrade` command, which should resolve this issue.

**Comment by Abhishek01039** [comment]
But I can not upgrade my flutter version. because we have very big project if I upgrade it then there are so much refactoring @hm21

**Comment by Abhishek01039** [comment]
can you please tell me which version of `pro_image_editor` works fine with flutter 3.19.6 @hm21 ?

**Comment by hm21** [comment]
I think it should be version `2.7.8` or lower. But keep in mind that if you downgrade so many versions, you will have some other problems and a big performance impact.

Alternatively, you can fork this repository and change this part [here](https://github.com/hm21/pro_image_editor/blob/90c848cf25232507025c2eb13f8306e58c9adc4c/lib/utils/content_recorder.dart/content_recorder_controller.dart#L407-L412) to the code below, which should also resolve it.

```dart
configuration: ViewConfiguration(
    size: logicalSize,
    devicePixelRatio: pixelRatio ?? 1.0,
  ),
```

**Comment by Abhishek01039** [comment]
I resolved this

if someone also faces the same issue please suggest them to use this in their pubspec.yaml file
```
pro_image_editor:
    git:
      url: https://github.com/Abhishek01039/pro_image_editor
      ref: flutter-3.19.6
```

Thanks for your quick responses @hm21

**Comment by Abhishek01039** [comment]
Hello @hm21 
I am facing this issue on GitHub action (CI/CD)

<img width="691" alt="image" src="https://github.com/hm21/pro_image_editor/assets/56479665/6b3c2235-1f12-47a8-85b6-fb7691c16549">

This is happening after adding this package

**Comment by Abhishek01039** [comment]
can you please also upgrade `wakelock` package to latest version?
<img width="1507" alt="image" src="https://github.com/hm21/pro_image_editor/assets/56479665/59ee3434-1a2b-44ed-8957-0cac8ee143ad">

**Comment by hm21** [comment]
Actually, I don't have `wakelock` in the `pubspec.yaml` file, but it seems to be from the `vibration` package. Anyway, all packages use the latest version and if we look at your error message, you should downgrade `wakelock` to version `0.4.0`. I think it will be fixed if you use an older version of the `vibration` package or remove it completely. Alternatively, you can change the version of `device_info_plus` as these two packages have a conflict.

FYI I released version `4.2.0` which now works differently how it captures widgets. I used before a code part from the `screenshot` package, but because it's not backward compatible and many people ask me they need it in older versions, I changed it to my own solution which should also work in older flutter versions.

**Comment by Abhishek01039** [comment]
the 4.2.0 fixed my issues. thanks!
